measures/comstock_sensitivity_reports/measure.rb

```diff
@@ -2719,55 +2719,107 @@ def run(runner, user_arguments)
     runner.registerValue('com_report_hvac_hot_water_loop_boiler_fraction', hot_water_loop_boiler_fraction)
     runner.registerValue('com_report_hvac_hot_water_loop_heat_pump_fraction', hot_water_loop_heat_pump_fraction)
 
-    # Average gas coil efficiency
-    gas_coil_capacity_weighted_efficiency = 0.0
-    gas_coil_total_capacity_w = 0.0
-    gas_count_0_to_30_kbtuh = 0.0
-    gas_count_30_to_65_kbtuh = 0.0
-    gas_count_65_to_135_kbtuh = 0.0
-    gas_count_135_to_240_kbtuh = 0.0
-    gas_count_240_plus_kbtuh = 0.0
+    # Average primary gas coil efficiency
+    primary_gas_coil_capacity_weighted_efficiency = 0.0
+    primary_gas_coil_total_capacity_w = 0.0
+    primary_gas_count_0_to_30_kbtuh = 0.0
+    primary_gas_count_30_to_65_kbtuh = 0.0
+    primary_gas_count_65_to_135_kbtuh = 0.0
+    primary_gas_count_135_to_240_kbtuh = 0.0
+    primary_gas_count_240_plus_kbtuh = 0.0
+
+    # Average supplemental gas coil efficiency
+    supplemental_gas_coil_capacity_weighted_efficiency = 0.0
+    supplemental_gas_coil_total_capacity_w = 0.0
+    supplemental_gas_count_0_to_30_kbtuh = 0.0
+    supplemental_gas_count_30_to_65_kbtuh = 0.0
+    supplemental_gas_count_65_to_135_kbtuh = 0.0
+    supplemental_gas_count_135_to_240_kbtuh = 0.0
+    supplemental_gas_count_240_plus_kbtuh = 0.0
+    
+    # iterate through each model to get all of the gas coils and check if they are supplemental coils for Unitary HVAC Objects and count each seperately.
     model.getCoilHeatingGass.sort.each do |coil|
       # get gas coil capacity
-      capacity_w = 0.0
-      if coil.nominalCapacity.is_initialized
-        capacity_w = coil.nominalCapacity.get
-      elsif coil.autosizedNominalCapacity.is_initialized
-        capacity_w = coil.autosizedNominalCapacity.get
+      supplemental_capacity_w = 0.0
+      if !model.getAirLoopHVACUnitarySystems.empty? 
```

Comment:
this selection criteria isn't going to work - 
1) some models may have secondary systems types that are AirLoopHVACUnitarySystems. If the primary system has gas coils, but has AirLoopHVACUnitarySystems, then all gas coils will be labeled as supplemental.
2) Some systems, particularly PSZ-AC RTUs, are modeled as AirLoopHVACUnitarySystems. They have primary gas coils, which will also be logged as supplementary here.

I suggest testing some models with these kinds of variations and seeing if the output variables are logging the capacities into the correct bins.

Reply:
perhaps the more direct method is to call .supplementalCoil and check if the coil name in the loop matches the coil on the systems (in essence, whether the coil is part of a unitary system, and is the supplementary coil on that system.

Reply:
Matt, I have made some updates to the code to use the .supplementalCoil method. We can discuss in person 3/12